openwisp_controller/config/tasks.py

```diff
@@ -0,0 +1,20 @@
+import logging
+
+from celery import shared_task
+from celery.exceptions import SoftTimeLimitExceeded
+from swapper import load_model
+
+logger = logging.getLogger(__name__)
+Vpn = load_model('config', 'Vpn')
+
+
+@shared_task(soft_time_limit=1200)
+def create_vpn_dhparam(vpn_pk):
+    vpn_obj = Vpn.objects.get(pk=vpn_pk)
+    try:
+        vpn_obj.dh = Vpn.dhparam(2048)
+    except SoftTimeLimitExceeded as e:
+        logger.warning(f'A timeout exception occurred when generating the DH: {e}')
```

Comment:
when > while, no need to add `e` to the output because it will be instance of `SoftTimeLimitExceeded` and I think won't add much value to the log line